scripts/install.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+ROOT_DIR="$(pwd)"
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    OS="macOS"
+    NUMBER_OF_PROCESSORS="$(sysctl -n hw.ncpu)"
+fi
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    OS="Linux"
+    NUMBER_OF_PROCESSORS="$(nproc)"
+fi
+
+GENERATOR="Unix Makefiles"
+
+SUBMODULE_DIR="${ROOT_DIR}/vendor"
+SUBMODULE_INCLUDE_DIR="${SUBMODULE_DIR}/include"
+SUBMODULE_LIB_DIR="${ROOT_DIR}/lib/${OS}"
+
+echo "Installing vulkan dependencies..."
+
+update_submodules() {
+    git submodule update --init ${SUBMODULE_DIR}/$1
+}
+
+mkdirp() { 
+    mkdir -p $1 
+}
+
+setup_volk() {
+    echo "Cloning Volk..."
+    update_submodules volk
+
+    mkdirp ${SUBMODULE_DIR}/include/volk
+
+    cp ${SUBMODULE_DIR}/volk/volk.h ${SUBMODULE_INCLUDE_DIR}/volk 
+    cp ${SUBMODULE_DIR}/volk/volk.c ${SUBMODULE_INCLUDE_DIR}/volk
```

Comment:
this is not a header?

Reply:
No. We don't build `volk.c` but we do need it in order to use `volk.h`.

Reply:
need it in what way?

Reply:
If we don't include it the build fails. I'm afraid this isn't just a header only library

Reply:
wait, how how does it get included/built??

Reply:
are we including `volk.c` somewhere, or are we compiling?

Reply:
Yes we do. It's one of the includes in our `Makefile` [here](https://github.com/Raelr/vk-sandbox/blob/577b685b63e1855c2fdadc5f193dc55517fb1c70/Makefile#L15)